include/mapbox/variant.hpp

```diff
@@ -588,32 +588,48 @@ class variant
 public:
     VARIANT_INLINE variant() noexcept(std::is_nothrow_default_constructible<first_type>::value)
         : type_index(sizeof...(Types)-1)
+#ifdef __clang_analyzer__
+        , data{}
```

Comment:
@julianrex - I thought we only need an explicit  `data{}` where it's initialised via `helper_type::move` and `helper_type::copy`, could confirm pls. If we do really need to call ctor on `data` to pacify static analyser then less intrusive solution could be in-class initialisation, something like : 

```c++
#ifdef __clang_analyzer__
      data_type data {};
#else  
     data_type data;
#endif 
```

Reply:
Pushed the change to in-class initialisation; it was needed for all constructors.